### PR TITLE
Delete the temporary file after reading

### PR DESCRIPTION
**Pull request by ph**
When the files are downloaded from S3 they were not deleted after reading.

### CODE REVIEW

File: lib/logstash/inputs/s3.rb [hunk 1]
```diff
@@ -293,6 +294,7 @@ def process_log(queue, key)
     backup_to_dir(filename)
 
     delete_file_from_bucket(object)
+    FileUtils.remove_entry_secure(filename, true)
```
Comment (colinsurprenant) [inline lib/logstash/inputs/s3.rb]: TIL

**Comment by ph** [comment]
@colinsurprenant Moved the duplicate test in the shared example

File: spec/inputs/s3_spec.rb [hunk 2]
```diff
@@ -160,6 +165,18 @@
     end
   end
 
+  shared_examples "temporary files"  do
+    it 'should process events' do
```
Comment (colinsurprenant) [inline spec/inputs/s3_spec.rb]: should the shared example still be called "temporary files" ?

**Comment by colinsurprenant** [comment]
LGTM :shipit:

**Pull request closed without merging**